service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -171,22 +173,24 @@ func (r *Resource) updateAppCatalogEntry(ctx context.Context, entryCR *v1alpha1.
 func (r *Resource) newAppCatalogEntries(ctx context.Context, cr v1alpha1.AppCatalog, index index) (map[string]*v1alpha1.AppCatalogEntry, error) {
 	entryCRs := map[string]*v1alpha1.AppCatalogEntry{}
 
-	for name, entries := range index.Entries {
-		sortedEntries, err := sortEntry(entries)
+	for _, entries := range index.Entries {
+		sort.Slice(entries, func(i, j int) bool {
+			return entries[i].Created.After(entries[j].Created.Time)
+		})
+
+		latestVersion, err := r.getLatestVersion(ctx, entries)
```

Comment:
It should be an edge case but what happens if the latest version is not in the max entries? Will there still be a CR for it?

Reply:
Oops, no, in these case, latest one would be not included. Do you think it makes sense to add `latest` always?

Reply:
So what about adding the latest `AppCatalogEntry` regardless the number of max entries?

Reply:
Yes I think always including latest makes sense. 

Otherwise if you filter by latest and its not there it would be weird for things like Happa when it eventually uses these CRs.